app/controllers/preprints/provider/moderators.js

```diff
@@ -0,0 +1,216 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+
+import QueryParams from 'ember-parachute';
+import { task } from 'ember-concurrency';
+
+import Analytics from 'ember-osf/mixins/analytics';
+
+
+export const moderatorsQueryParams = new QueryParams({
+    page: {
+        defaultValue: 1,
+        refresh: true,
+    },
+});
+
+export default Controller.extend(Analytics, moderatorsQueryParams.Mixin, {
+    store: service(),
+    theme: service(),
+    i18n: service(),
+    currentUser: service(),
+
+    disableAdminDeletion: false,
+    editingModerator: false,
+    addingNewContrib: false,
+    isAdmin: false,
+
+    actions: {
+        pageChanged(page) {
+            this.set('page', page);
+        },
+        newModerator() {
+            this.setProperties({
+                editingModerator: true,
+                addingNewModerator: true,
+            });
+            if (this.get('page') !== this.get('results.totalPages')) {
+                this.set('page', this.get('results.totalPages'));
+            }
+        },
+    },
+
+    setup({ queryParams }) {
+        this.set('moderatorIds', []);
+        this.set('roleOptions', [
+            {
+                role: 'admin',
+                label: 'Admin',
+            },
+            {
+                role: 'moderator',
+                label: 'Moderator',
+            },
+        ]);
+        this.get('fetchAdmin').perform();
+        this.get('fetchData').perform(queryParams);
+        this.get('loadModerators').perform();
+    },
+
+    queryParamsDidChange({ shouldRefresh, queryParams }) {
+        if (shouldRefresh) {
+            this.get('fetchData').perform(queryParams);
+        }
+    },
+
+    reset(isExiting) {
+        if (isExiting) {
+            this.resetQueryParams();
+        }
+    },
+
+    loadModerators: task(function* () {
+        const moderators = yield this.get('store').query('moderator', {
+            page: {
+                size: 100,
+            },
+            provider: this.get('theme.provider.id'),
+        });
+        const moderatorIds = moderators.map(moderator => moderator.id);
+        this.set('moderatorIds', moderatorIds);
+    }),
+
+    deleteModerator: task(function* (id) {
+        try {
+            // queryRecord doesn't play nice with the API
+            // findRecord with delete won't accept a payload
```

Comment:
Could you pass `moderator` instead of `moderator.id` and do `moderator.destroyRecord()` right off the bat? Or is it important to double-check that it wasn't already deleted?

Reply:
I'm not sure if I tried that -- fingers crossed 🤞